src/resources/tls.cc

```diff
@@ -702,5 +714,81 @@ PRIMITIVE(set_session) {
   return process->program()->null_object();
 }
 
+PRIMITIVE(get_internals) {
+  ARGS(BaseMbedTlsSocket, socket);
+  size_t iv_len = socket->ssl.transform_out->ivlen;
+  // mbedtls_cipher_context_t from include/mbedtls/cipher.h.
+  mbedtls_cipher_context_t* out_cipher_ctx = &socket->ssl.transform_out->cipher_ctx_enc;
+  mbedtls_cipher_context_t* in_cipher_ctx = &socket->ssl.transform_in->cipher_ctx_dec;
+  size_t key_bitlen = out_cipher_ctx->key_bitlen;
+  // mbedtls_cipher_info_t from include/mbedtls/cipher.h.
+  const mbedtls_cipher_info_t* out_info = out_cipher_ctx->cipher_info;
+  const mbedtls_cipher_info_t* in_info = in_cipher_ctx->cipher_info;
+
+  // Sanity check the connection for parameters we can cope with.
```

Comment:
I think this would be easier to work with if you put it in a static predicate function. Early returns instead of all these ||.